services/brig/src/Brig/API.hs

```diff
@@ -1037,6 +1049,14 @@ internalListClients (_ ::: _ ::: req) = do
     ucs <- Map.fromList <$> lift (API.lookupUsersClientIds $ Set.toList usrs)
     return $ json (UserClients ucs)
 
+internalQueryServiceWhitelist :: TeamId ::: Request -> Handler Response
+internalQueryServiceWhitelist (tid ::: req) = do
+    svcs :: [ServiceRef] <- parseJsonBody req
+    fmap json $ forM svcs $ \svc -> do
+        let pid = svc ^. serviceRefProvider
+            sid = svc ^. serviceRefId
+        getServiceWhitelistStatus tid pid sid
+
```

Comment:
Given that you want to get the whitelisted services for a single team, you can do this in a single read; with the definition:
```
        create table if not exists service_whitelist
            ( team     uuid
            , provider uuid
            , service  uuid
            , primary key (team, provider, service)
            ) with clustering order by (provider asc, service asc)
```
you can simply do `SELECT pid,sid FROM service_whitelist` and then filter it at the application level.

Actually, assuming that there are only a handful services whitelisted, the most efficient could even be to implement a `GET /teams/:id/services/whitelisted` or something like that although this is something that I don't _really_ suggest :)